### ISSUE TITLE
Use multi config files and obervatory gets "panic: runtime error: invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by ha-ku** [v2fly/v2ray-core#944]
## What version of V2Ray are you using?

server: v4.37.3
client: v4.38.2

## What's your scenario of using V2Ray?

Watching YouTube videos in browser via websocket + TLS VMess proxy using observatory module with seperate multiple json config file

## What problems have you encountered?

**Only when using seperate config files**, client immediately panic **only after request send to balancer using leastPing stategy**, showing:
panic: runtime error: invalid memory address or nil pointer dereference

I merge the config files by running
```
echo {} | jq --argfile in inbounds.json --argfile out outbounds.json --argfile rt routing.json --argfile tr transport.json --argfile bf browserForwarder.json  --argfile ob observatory.json '$in + $out + $rt + $tr + $ bf + $ob' | sudo tee config.json > /dev/null
```
and `v2ray -config config.json` works fine, which is weird. 

## What's your expectation?
Both single config file and multiple config files can work fine.

## Please attach your configuration here

**Server configuration:**

```javascript
// Please attach your server configuration here.

```

**Client configuration:**

inbounds.json
```javascript
{
  "inbounds": [
    {
      "listen": "0.0.0.0",
      "port": 31081,
      "protocol": "http",
      "settings": {
        "timeout": 0
      },
      "tag": "cdn-http"
    }
  ]
}
```

outbounds.json
```javascript
{
  "outbounds": [
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "xxx.xxxx.xxxx",
            "port": 443,
            "users": [
              {
                "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
                "alterId": 0,
                "security": "aes-128-gcm"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls"
      },
      "tag": "abc-cdn"
    },
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "xxxxxxx.xxxx.xxxx",
            "port": 443,
            "users": [
              {
                "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
                "alterId": 0,
                "security": "aes-128-gcm"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls"
      },
      "tag": "def-cdn"
    },
        {
            "protocol": "blackhole"
            ,"tag": "block"
        }
        ,{
            "protocol": "freedom"
            ,"domainStrategy": "UseIP"
            ,"tag": "direct"
        }
  ]
}
```

routing.json
```javascript
{
  "routing": {
    "domainStrategy": "IPIfNonMatch",
    "domainMatcher": "mph",
    "balancers": [
      {
        "tag": "lb-cdn",
        "selector": [
          "abc-cdn",
          "def-cdn"
        ],
        "strategy": {
          "type": "random"
        }
      },
      {
        "tag": "lantency-cdn",
        "selector": [
          "abc-cdn",
          "def-cdn"
        ],
        "strategy": {
          "type": "leastPing"
        }
      }
    ],
    "rules": [
      {
        "type": "field",
        "domains": [
          "domain:youtubei.googleapis.com",
          "domain:spotify.com",
          "domain:youtube.com",
          "domain:googlevideo.com"
        ],
        "balancerTag": "lantency-cdn"
      },
      {
        "type": "field",
        "network": "tcp,udp",
        "balancerTag": "lb-cdn"
      }
    ]
  }
}
```

transport.json
```javascript
{
  "transport": {
    "wsSettings": {
      "path": "/xxxxx",
      "maxEarlyData": 1024,
      "useBrowserForwarding": false
    }
  }
}
```

observatory.json
```javascript
{
  "observatory": {
    "subjectSelector": [
      "abc",
      "def"
    ]
  }
}
```

log.json
```javascript
{
    "log": {
                "loglevel": "debug"
    }
}
```
## Please attach error logs here

**Server error log:**

```javascript
// Please attach your server error log here.
// empty
```

**Client error log:**

```javascript
// Please attach your client error log here.
V2Ray 4.38.2 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.16.3 linux/arm64)
A unified platform for anti-censorship.
2021/04/30 05:24:20 Using confdir from arg: /usr/local/etc/v2ray/
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/inbounds.json
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/log.json
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/observatory.json
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/outbounds.json
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/routing.json
2021/04/30 05:24:20 [Info] main/jsonem: Reading config: /usr/local/etc/v2ray/transport.json
2021/04/30 05:24:20 [Debug] app/log: Logger started
2021/04/30 05:24:20 [Debug] app/router: MphDomainMatcher is enabled for 4 domain rule(s)
2021/04/30 05:24:20 [Debug] app/proxyman/inbound: creating stream worker on 0.0.0.0:31081
2021/04/30 05:24:20 [Info] transport/internet/tcp: listening TCP on 0.0.0.0:31081
2021/04/30 05:24:20 [Warning] V2Ray 4.38.2 started
```
(run following command on another shell: curl -x http://localhost:31081 https://www.googlevideo.com)
```
2021/04/30 05:24:30 [Info] [318963911] proxy/http: request to Method [CONNECT] Host [www.googlevideo.com:443] with URL [//www.googlevideo.com:443]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x577260]

goroutine 5 [running]:
github.com/v2fly/v2ray-core/v4/app/router.(*LeastPingStrategy).PickOutbound(0x40005365a0, 0x4000536000, 0x2, 0x2, 0x4000536000, 0x2)
        github.com/v2fly/v2ray-core/v4/app/router/strategy_leastping.go:32 +0x30
github.com/v2fly/v2ray-core/v4/app/router.(*Balancer).PickOutbound(0x40005095c0, 0xb15848, 0x400000c030, 0x4000544c30, 0xb15848)
        github.com/v2fly/v2ray-core/v4/app/router/balancing.go:44 +0x16c
github.com/v2fly/v2ray-core/v4/app/router.(*Rule).GetTag(...)
        github.com/v2fly/v2ray-core/v4/app/router/config.go:58
github.com/v2fly/v2ray-core/v4/app/router.(*Router).PickRoute(0x4000509540, 0xb15848, 0x400000c030, 0x400000c030, 0x0, 0x77024, 0x4000525068)
        github.com/v2fly/v2ray-core/v4/app/router/router.go:78 +0x134
github.com/v2fly/v2ray-core/v4/app/dispatcher.(*DefaultDispatcher).routedDispatch(0x4000509440, 0xb0b7c0, 0x40000302d0, 0x400003e060, 0xb0ca90, 0x400021e400, 0x2006001bb)
        github.com/v2fly/v2ray-core/v4/app/dispatcher/default.go:314 +0x338
created by github.com/v2fly/v2ray-core/v4/app/dispatcher.(*DefaultDispatcher).Dispatch
        github.com/v2fly/v2ray-core/v4/app/dispatcher/default.go:217 +0x570
```
(curl return `curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to www.googlevideo.com:443`)
## Please attach access log here

```javascript
// Please attach your server access log here.
// empty
```

## If V2Ray cannot start up, please attach output from `--test` command
Test shows configure OK

**Comment by xiaokangwang** [comment]
observatory is not currently supported by muti-json. You can set observatory in the first json configure file to workaround this.

**Comment by ha-ku** [comment]
> observatory is not currently supported by muti-json. You can set observatory in the first json configure file to workaround this.

This solution works for me. I run `mv observatory.json 00observatory.json`. 
By the way, I found the browserForwarder Object seems also conflicts with multi-json. I checked the overide function and found a similar part of code seems missed. Meanwhile, the workaround you mentioned above also works for this. Since I know little about go, I didn't make a PR. Maybe you would like to do some further investigation?
